pyt/cfg/expr_visitor.py

```diff
@@ -113,22 +114,26 @@ def visit_Yield(self, node):
         label = LabelVisitor()
         label.visit(node)
 
-        try:
-            rhs_visitor = RHSVisitor()
-            rhs_visitor.visit(node.value)
-        except AttributeError:
-            rhs_visitor.result = 'EmptyYield'
+        if node.value is None:
+            rhs_visitor_result = []
+        else:
+            rhs_visitor_result = RHSVisitor.result_for_node(node.value)
 
+        # Yield is a bit like augmented assignment to a return value
         this_function_name = self.function_return_stack[-1]
-        LHS = 'yield_' + this_function_name
-        return self.append_node(ReturnNode(
-            LHS + ' = ' + label.result,
+        LHS = 'yld_' + this_function_name
+        return self.append_node(YieldNode(
+            LHS + ' += ' + label.result,
             LHS,
             node,
-            rhs_visitor.result,
-            path=self.filenames[-1])
+            rhs_visitor_result + [LHS],
+            path=self.filenames[-1],
+            line_number=node.lineno)
```

Comment:
So for this I don't think you need to pass `line_number` due to
https://github.com/python-security/pyt/blob/master/pyt/core/node_types.py#L42-L43